### PR TITLE
Add folder delete and rename behavior

### PR DESCRIPTION
**Pull request by benjreinhart**
Need to fix folder renaming, but otherwise this is good to go.

### CODE REVIEW

File: packages/api/server/http.mts [hunk 1]
```diff
@@ -515,6 +517,51 @@ router.post('/apps/:id/directories', cors(), async (req, res) => {
   }
 });
 
+router.options('/apps/:id/directories', cors());
+router.delete('/apps/:id/directories', cors(), async (req, res) => {
+  const { id } = req.params;
+
+  // TODO: validate and ensure path is not absolute
```
Comment (nichochar) [inline packages/api/server/http.mts:524]: I've wanted some path cleaning utils also in other places, this is a good idea. It's particularly useful given that LLMs are giving us paths which i have less confidence in.

File: packages/web/src/components/apps/use-files.tsx [hunk 2]
```diff
@@ -183,6 +209,49 @@ export function FilesProvider({ app, channel, rootDirEntries, children }: Provid
...
+      fileTreeRef.current = deleteNode(fileTreeRef.current, entry.path);
+      forceComponentRerender(); // required
+    },
+    [app.id],
+  );
+
+  const renameFolder = useCallback(
+    async (entry: DirEntryType, name: string) => {
+      const { data: newEntry } = await renameDirectory(app.id, entry.path, name);
+      // const oldFile = filesRef.current[entry.path];
```
Comment (nichochar) [inline packages/web/src/components/apps/use-files.tsx:232]: Do we need to keep this?

**Review by nichochar — APPROVED** [review]
LGTM

File: packages/web/src/components/apps/use-files.tsx (hunk 2 above)
Comment (benjreinhart) [reply packages/web/src/components/apps/use-files.tsx:232]: This is what I need to fix before merging

**Pull request merged**